### PR TITLE
Hip interval to metababel

### PR DESCRIPTION
**Pull request by DonAurelio**
* Migrate hip interval to metababel.
* Update the hip_model generation script to include `common_context`,  `package_context`, and `environment`. Since the script `babeltrace_to_metababel_model` require those fields to be defined.
* Create a new babeltrace_to_metababel_model script to parse the old model to the metababel version.
* Delete `clock_snapshot_value` keyword from the xprof/interval_model.yaml. Since the interval events do not have a clock.

### CODE REVIEW

**Comment by TApplencourt** [comment]
>Stil working on the Makefile, distcheck still not working :(

Haha, got beaten by that many time :) We can have a look together on Monday. Look like you forgot add 

I think, the CI doesn't pass because the gemm have not updated.

File: hip/btx_hipinterval_callbacks.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#include <regex>
+#include <tuple>
+
+#include <hip.h.include>
+
+#include "xprof_utils.hpp"
+#include "btx_hipinterval_callbacks.hpp"
+
+
+void send_message_lttng_host(void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name,
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:15]:
```suggestion
static void send_host_message(void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name, 
```

File: hip/btx_hipinterval_callbacks.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#include "xprof_utils.hpp"
+#include "btx_hipinterval_callbacks.hpp"
+
+
+void send_message_lttng_host(void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name, 
+                             const char *hostname, int64_t vpid, uint64_t vtid, int64_t err){
+  auto *data = static_cast<data_t *>(usr_data);
+
+  std::string event_class_name_striped = strip_event_class_name(event_class_name);
+  int64_t _start = data->dispatch.at(hpt_fn_t(hostname, vpid, vtid, event_class_name_striped));
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:20]:
```suggestion
  const int64_t _start = data->dispatch.at({hostname, vpid, vtid, event_class_name_striped});
```

File: hip/btx_hipinterval_callbacks.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                              BACKEND_HIP, event_class_name_striped.c_str(), (ts - _start), err);                     
+}
+
+void btx_initialize_component(void *btx_handle, void **usr_data) {
+  *usr_data = new data_t;
+}
+
+void btx_finalize_component(void *btx_handle, void *usr_data) {
+  auto *data = static_cast<data_t *>(usr_data);
+  delete data;
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:32]:
```suggestion
  delete static_cast<data_t *>(usr_data);
```

File: hip/btx_hipinterval_callbacks.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+static void entries_callback(void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name, 
+                             const char *hostname, int64_t vpid, uint64_t vtid) {
+  data_t *data = (data_t *)usr_data;
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:37]:
```suggestion
  auto *data = static_cast<data_t *>usr_data;
```

File: hip/btx_hipinterval_callbacks.cpp [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  data->dispatch[hpt_fn_t(hostname, vpid, vtid, strip_event_class_name(event_class_name))] = ts;
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:38]:
```suggestion
  data->dispatch[{hostname, vpid, vtid, strip_event_class_name(event_class_name)}] = ts;
```

File: hip/btx_hipinterval_callbacks.cpp [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+static void exits_callback_hipError_absent(
+  void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name, 
+  const char *hostname, int64_t vpid, uint64_t vtid) {
+
+  send_message_lttng_host(btx_handle, usr_data, ts, event_class_name, hostname, vpid, vtid, 0);
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:45]:
```suggestion
  send_host_message(btx_handle, usr_data, ts, event_class_name, hostname, vpid, vtid, false);
```

File: hip/btx_hipinterval_callbacks.cpp [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+static void exits_callback_hipError_present(
+  void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name, 
+  const char *hostname, int64_t vpid, uint64_t vtid, int64_t hipResult) {
+
+  // Not an Error (hipResult == hipErrorNotReady)
+  int64_t err = hipResult != 0 ? (hipResult == hipErrorNotReady ? 0 : hipResult) : 0; 
+  send_message_lttng_host(btx_handle, usr_data, ts, event_class_name, hostname, vpid, vtid, err);
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:54]:
```suggestion
  send_host_message(btx_handle, usr_data, ts, event_class_name, hostname, vpid, vtid, err);
```

File: hip/btx_hipinterval_callbacks.cpp (hunk 6 above)
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:45]:
```suggestion
  send_host_message(btx_handle, usr_data, ts, event_class_name, hostname, vpid, vtid, false);
```

File: hip/btx_hipinterval_callbacks.cpp [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+#include <tuple>
+
+#include <hip.h.include>
+
+#include "xprof_utils.hpp"
+#include "btx_hipinterval_callbacks.hpp"
+
+
+void send_message_lttng_host(void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name, 
+                             const char *hostname, int64_t vpid, uint64_t vtid, int64_t err){
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:16]:
```suggestion
                             const char *hostname, int64_t vpid, uint64_t vtid, bool err){
```

File: hip/btx_hipinterval_callbacks.cpp [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+  send_message_lttng_host(btx_handle, usr_data, ts, event_class_name, hostname, vpid, vtid, 0);
+}
+
+static void exits_callback_hipError_present(
+  void *btx_handle, void *usr_data, int64_t ts, const char *event_class_name, 
+  const char *hostname, int64_t vpid, uint64_t vtid, int64_t hipResult) {
+
+  // Not an Error (hipResult == hipErrorNotReady)
+  int64_t err = hipResult != 0 ? (hipResult == hipErrorNotReady ? 0 : hipResult) : 0;
```
Comment (DonAurelio) [inline hip/btx_hipinterval_callbacks.cpp:53]:
```suggestion
  bool err = (hipResult != 0) && (hipResult != hipErrorNotReady);
```

**Comment by DonAurelio** [comment]
I think we are goo now.

**Comment by DonAurelio** [comment]
I think we are good to go.

**Comment by DonAurelio** [comment]
We can void the `$(srcdir)` in the dependencies since `btx_interval_model.yaml` and `btx_timeline_params.yaml` are in EXTRA_DIST, that's why it works before. This tells Make that those dependencies exist and we need to copy them for distribution (only when `make dist` is issued, not during build).

Then, it is mandatory to place `$(srcdir)` in the command since those files won't be present in the build dir when we issue `make`. I chose to keep $(srcdir)` in the dependencies to tell this dependencies are not in the build directory at build time (`make`). What do you think?

```make
$(BTX_TIMELINE_GENERATED) &: $(srcdir)/btx_interval_model.yaml $(srcdir)/btx_timeline_params.yaml
	$(METABABEL) -u $(srcdir)/btx_interval_model.yaml -t SINK -o btx_sink_timeline -p btx_timeline -c timeline --params $(srcdir)/btx_timeline_params.yaml

```

**Review by TApplencourt — APPROVED** [review]

**Comment by TApplencourt** [comment]
Good enough, it was not the main of this PR :) I just merged it, so we can move forward :)

**Pull request merged**